localstack-core/localstack/testing/scenario/provisioning.py

```diff
@@ -243,9 +243,10 @@ def provision(self, skip_deployment: Optional[bool] = False):
                 ]
 
                 for s3_bucket in s3_buckets:
+                    bucket_to_delete = s3_bucket  # Capture current value
                     self.custom_cleanup_steps.append(
-                        lambda: cleanup_s3_bucket(
-                            self.aws_client.s3, s3_bucket, delete_bucket=False
+                        lambda bucket=bucket_to_delete: cleanup_s3_bucket(
+                            self.aws_client.s3, bucket, delete_bucket=False
```

Comment:
```suggestion
                        lambda s3_bucket=s3_bucket: cleanup_s3_bucket(
                            self.aws_client.s3, s3_bucket, delete_bucket=False
```

this alone would be enough to bind the value here

Reply:
I've update the `lambda` to use the loop variable.